docs/source/write_first_app.rst

```diff
@@ -101,889 +97,485 @@ bringing up a new one.
-identities for two separate users --- ``admin`` and ``appUser`` --- that can be
-used by our application.
+  1. gRPC connection to the Gateway peer.
+  2. Client identity used to transact with the network.
+  3. Signing implementation used to generate digital signatures for the client identity.
 
-Scrolling further down in your terminal output, you should see confirmation of the app user registration
-similar to this:
+In addition to the above, a hash implementation can also be specified. The default implementation will typically be
+sufficient.
 
-.. code:: bash
+The sample application uses an X.509 certificate for the Org1 user identity, and a signing implementation based on the
+corresponding private key. These credentials are created using the Org1 Certificate Authority when the test network
+is started.
 
-  Successfully registered and enrolled user appUser and imported it into the wallet
+.. code-block:: TypeScript
 
-Third, the sample application prepares a connection to the channel and smart contract
--------------------------------------------------------------------------------------
+  const client = await newGrpcConnection();
 
-In the prior steps, the application generated the admin and app user credentials and placed them in the wallet.
-If the credentials exist and have the correct permissions attributes associated with them, the sample application user
-will be able to call chaincode functions after getting reference to the channel name and contract name.
+  const gateway = connect({
+      client,
+      identity: await newIdentity(),
+      signer: await newSigner(),
+  });
 
-.. note:: Our connection configuration specifies only the peer from your own Org.
-          We tell node client sdk to use the service discovery (running on the peer),
-          which fetches other peers that are currently online, metadata like relevant endorsement policies
-          and any static information it would have otherwise needed to communicate with the rest of the nodes.
-          The ``asLocalhost`` set to ``true`` tells it to connect as localhost, since our client is running on same network as the other fabric nodes.
-          In deployments where you are not running the client on the same network as the other fabric nodes,
-          the ``asLocalhost`` option would be set to ``false``.
+  async function newIdentity(): Promise<Identity> {
+      const credentials = await fs.readFile(certPath);
+      return { mspId: 'Org1MSP', credentials };
+  }
 
-You will notice that in the following lines of application code, the application is getting reference
-to the Contract using the contract name and channel name via Gateway:
+  async function newSigner(): Promise<Signer> {
+      const privateKeyPem = await fs.readFile(keyPath);
+      const privateKey = crypto.createPrivateKey(privateKeyPem);
+      return signers.newPrivateKeySigner(privateKey);
+  }
 
-.. code:: bash
+Third, the application accesses the smart contract to be invoked
```

Comment:
Okay so this is the application flow right, for gateway. I think there are 3 phases in the other gateway pages. Are these related to your 3 steps?

Reply:
No idea. These are the steps from the application perspective when using the client API.